packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -566,6 +566,7 @@ class _FocusState extends State<Focus> {
   void dispose() {
     // Regardless of the node owner, we need to remove it from the tree and stop
     // listening to it.
+    widget.onFocusChange?.call(false);
```

Comment:
I see. 

> Unfortunately, I'm kind of at a loss here—not sure how we could solve this issue without breaking things 🙁
Do you think we should close this PR? If so what could be done with the issue? 

Just sharing some thoughts about the line below.
`focusNode.removeListener(_handleFocusChanged);`

it sounds like when we trigger the `_handleFocusChanged` the listener was already clean up. So the callback inside it won't work.
And even if it worked the focusNode is being removed so its value is not being updated.

It makes me wonder if it is a bug or just the regular behavior.

Reply:
@gspencergoog WDYT? Does this sound like the right behavior?

Reply:
You don't need to call this explicitly. If you want to have that happen, just move the next line (the `removeListener` line) down one to below the `detach`, since detaching the attachment will force the node to give up focus and if the listener is still registered, it'll call it. You'd also be bypassing some code that runs in `_handleFocusChanged` if you call `widget.onFocusChange` directly.

However, I think that will cause a bunch of handlers to get called during dispose when they aren't expecting to, and they'll try to call `setState` and throw.

You could try and clean up those instances in the framework and see how that works for you, but it feels like something that will break a lot of code out there.

Reply:
Thanks for your help! 
I moved the `removeListener` below the `detach`, but it seems like I'm still getting the same result. When the widget is removed, the focus change doesn't trigger. 
By the time we reach the line `widget.onFocusChange?.call(focusNode.hasFocus);`, `widget.onFocusChange` is already null.

Reply:
Ahh, yeah, actually that makes sense.  The unfocus isn't going to actually take effect until after the current frame is done, so by that time the widget has already been disposed.

Hmm. Well, I guess you could try and call `widget.onFocusChange` directly here, but I still think that will cause problems. No harm in trying it, though. I would only call it if the node currently has focus.

Reply:
There's still something odd happening. The dispose method is being triggered twice. Is this expected behavior?

During the first call, focusNode.hasPrimaryFocus is false, and widget.onFocusChange is null.
During the second call, focusNode.hasPrimaryFocus is false, and widget.onFocusChange is `not` null.

if the following conditional I was able to make it work:
```
if (!focusNode.hasFocus && widget.onFocusChange != null) {
      widget.onFocusChange?.call(false);
    }
```

I didn't see any exceptions for now.